### PR TITLE
Adding a new auto-generated layer system.

### PR DESCRIPTION
**Pull request by AWoloszyn**
This allows a simpler way to create a vulkan layer.
This also updates the version of vulkan headers.

### CODE REVIEW

**Comment by hevrard** [comment]
@AWoloszyn this is a +72k line commit, can you clarify that you did not add some files by mistake? (e.g. generated files?)

**Comment by AWoloszyn** [comment]
You can ignore everything in the `core/vulkan/cc/include/vulkan` folder. This just revs the version of the Vulkan headers that we include.

File: kokoro/presubmit/presubmit.sh [hunk 1]
```diff
@@ -51,7 +51,7 @@ function check() {
 }
 
 function run_clang_format() {
-  find . -name "*.h" -o -name "*.cpp" -o -name "*.mm" -o -name "*.proto" | xargs $CLANG_FORMAT -i -style=Google
+  find . -path ./core/vulkan/cc/include/vulkan -prune -o \( -name "*.h" -o -name "*.cpp" -o -name "*.mm" -o -name "*.proto" \) -print| xargs $CLANG_FORMAT -i -style=Google
```
Comment (hevrard) [inline kokoro/presubmit/presubmit.sh]: Add a small comment above, e.g.:
`# Do not format files under core/vulkan/cc/include/vulkan/`

File: gapis/api/templates/vulkan_gfx_api_extras.tmpl [hunk 2]
```diff
@@ -389,7 +389,6 @@ bool Vulkan::replayCreateVkDeviceImpl(Stack* stack, size_val physicalDevice,
               break;
 
             // These are all linux, so just make an Xcb surface
-            case VK_STRUCTURE_TYPE_MIR_SURFACE_CREATE_INFO_KHR:
```
[the remark this answers is not in the document]
Comment (AWoloszyn) [reply gapis/api/templates/vulkan_gfx_api_extras.tmpl]: Yes MIR no longer exists in the latest vulkan headers.

**Pull request merged**